### PR TITLE
Allow tilt input on Z instead of X

### LINKED ISSUE #10515: Request: Android : Both L and R buttons pushed when tilted back
Add a little feature so that both L and R buttons get pushed when the device is tilted back and/or at least change the behaviour of the "tilt" mode so it doesn't block the touch input of the other button (Right now when it's on tilt mode if you tilt it to say left it doesn't allow you to touch the Right button

Playing GoW would be much easier that way

### LINKED ISSUE #12620: Conflict between tilt input and other input methods on Android in LocoRoco 2?
### What happens?

If you enable tilt input in Android for the **L** and **R** buttons in LocoRoco 2, and then set up a Combo Key or other method to press both L and R at once, that other input method doesn't work.

### What *should* happen?

LocoRoco uses the L and R buttons to tilt the world.  The game also has you press both L and R at once to shake the world to make your LocoRoco bounce.  These controls are the most important controls in the game.  Although PPSSPP allows you to control L and R with the acceleroeteer and LocoRoco handles this extremely well, there's no way to press both L and R at once.  A combo key or physical controller should be able to do it, but neither work.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

So far I've tested on an Alcatel A30 8-inch tablet running Android Nougat 7.1.1.  Not sure what PPSSPP version it's running but I got it from the Play Store.  Tablet's not with me right now but when I get home in a few hours, I'll update this with the version.  I'm also going to test on an LG G5 running Android Oreo with the latest Play Store version to see if the bug persists on there.

As for my input configuration, I have tilt input set to the L and R buttons.  This works as expected - tilting the device to the right, tilts the LocoRoco world to the right.  I have Combo Key 1 set to press L and R at once.  This should shake the LocoRoco world, but the combo key never registers a press when I tap it.  The game never sees the input.  I've also tried with an Xbox One S controller over Bluetooth, all inputs work fine except for the L and R buttons when the tilt control is on.  Perhaps the accelerometer is conflicting with other inputs?

### Further (but possibly not pertainant) information

I wanted to see if I could recreate a similar input conflict on desktop by hooking up a Logitech racing wheel to PPSSPP and setting the wheel up to control L and R (steering left should tilt the LocoRoco world to the left, steering right should tilt the world to the right.) Minus the fact I have to steer EXCESSIVELY to register a press, this works fine.  I also bound the rear buttons of the wheel to both press L and R at once so that either hand can bounce the world.  This also works with no issue, other than when you steer in a direction and press one of those buttons at the same time to shake while tilting.  This confuses the game and causes the world to flatten even though one of the tilt buttons is still held down, but I imagine that's a LocoRoco engine issue and not a PPSSPP issue.

### PR DESCRIPTION
**Pull request by iota97**
By using Z accelerometer axis it's possible to play when the phone screen is not parallel to the ground but rather perpendicular to it, making it easier to controll it from standing or sitting position.

I am totally lost about how to name such an option tho'.

Edit: Fixes #10515 and fixes #12620

### CODE REVIEW

**Comment by hrydgard** [comment]
Yeah, makes sense. Also not sure how to name it though :)

**Comment by i30817** [comment]
Why name it? Couldn't this be automatic? I don't understand why can't the axis for movements always be relative to the direction the screen is facing.

**Comment by iota97** [comment]
Added a third option (auto-switch) that change mode if it detect a big change in orientation, it seem to work fine on my phone, also having a multi-choice seem better at explaining what the setting does.

**Comment by iota97** [comment]
Fix #10515

**Comment by hrydgard** [comment]
Anything else you want to squeeze in here? I'm inclined to just go ahead and merge this, it's useful functionality.

**Comment by iota97** [comment]
The button masking is not perfect: it allow you to press a button controlled by tilt with touch as well, but it can be released by tilt even if it was pressed with touch in first place.
To handle all the possibility we will need some more masking variable, but i am not sure if is really worth the extra code complexity for such a corner case.

That the only thing i think it's left in case is needed.

**Comment by iota97** [comment]
Now it will not reset a button that was held when the tilt enable and then disable it.

If a button get pressed when it was also enabled by tilt, tilting back will release it even if the button is still hold on touch screen. 
I don't think this can be worked around easily with the current data struct (I might be very wrong on this tho'). I am not sure if is worth making thing more complex for such a case as well. Any suggestion is welcome ofc.

That's the last part that itches me, if you think is not worth going around that you may just go ahead and merge.

File: UI/TiltAnalogSettingsScreen.cpp [hunk 1]
```diff
@@ -35,6 +35,8 @@ void TiltAnalogSettingsScreen::CreateViews() {
 	settings->Add(new ItemHeader(co->T("Invert Axes")));
 	settings->Add(new CheckBox(&g_Config.bInvertTiltX, co->T("Invert Tilt along X axis")));
 	settings->Add(new CheckBox(&g_Config.bInvertTiltY, co->T("Invert Tilt along Y axis")));
+	static const char* tiltMode[] = { "Screen parallel to ground", "Screen orthogonal to ground", "Auto-switch" };
```
Comment (unknownbrackets) [inline UI/TiltAnalogSettingsScreen.cpp]: I know a bunch of people who have trouble with terms like orthogonal (or perpendicular) and parallel.  Not sure what's better, maybe "Screen level with ground" and "Screen upright from ground"?  Still seems confusing.

-[Unknown]

**Review by unknownbrackets — COMMENTED** [review]
I think if we end up counting the buttons pressing/not pressing inputs, we'll end up with bugs to fight when things are released/etc.  That doesn't necessarily mean we shouldn't, but it's definitely a road of complexity.

It looks like this at least doesn't release buttons that are held before and after the tilt.

I guess even combo buttons have this problem, in theory.  Haven't really seen people complain about that though, so maybe it's not as much of a problem...

-[Unknown]

File: UI/TiltAnalogSettingsScreen.cpp (hunk 1 above)
Comment (iota97) [reply UI/TiltAnalogSettingsScreen.cpp]: I find level and upright more confusing personally, but I guess it's a matter of native language.
What about "horizontal" and "vertical"? Might be a confused with landscape/portrait tho'.

Comment (unknownbrackets) [reply UI/TiltAnalogSettingsScreen.cpp]: Yeah, colloquial English just doesn't cover this well.  I guess people will figure it out.

-[Unknown]

Comment (vnctdj) [reply UI/TiltAnalogSettingsScreen.cpp]: I think "perpendicular" would be easier to understand than "orthogonal".

Comment (iota97) [reply UI/TiltAnalogSettingsScreen.cpp]: "Perpendicular" it's more common indeed, I wonder if "Screen aligned to ground" and "Screen at right angle to ground" would be simpler.

I can't quite tell what is easy and what is hard with words: some english word with a latin root are unusual to someone from USA or UK, but at the same time look almost obvious in Italy, Spain or France as they sound alike the everyday one in those languages.

Comment (unknownbrackets) [reply UI/TiltAnalogSettingsScreen.cpp]: I like those last two suggestions myself.  They're a bit long but "right angle to ground" would make sense to the vast majority of English speakers I know.  The other might not make sense to as many immediately, but presented with both they'd be obvious.

-[Unknown]

**Pull request merged**